### PR TITLE
Fix/improve handling of chdir with no-args and testpaths

### PR DESCRIPTION
**Pull request by blueyed**
Fixes https://github.com/pytest-dev/pytest/issues/4332.

### CODE REVIEW

**Comment by blueyed** [comment]
As per https://github.com/pytest-dev/pytest/pull/4317/files#r231699560 there could also be a warning, but that would add an extra `os.getcwd()` (not too bad though), but I am not really sure how helpful this would be after all.

File: src/_pytest/config/__init__.py [hunk 1]
```diff
@@ -848,10 +847,13 @@ def parse(self, args, addopts=True):
                 args, self.option, namespace=self.option
             )
             if not args:
-                if self.cwd == self.rootdir:
-                    args = self.getini("testpaths")
+                if self.invocation_dir == self.rootdir:
+                    args = [
+                        str(self.invocation_dir.join(x)) if not os.path.isabs(x) else x
```
Comment (RonnyPfannschmidt) [inline src/_pytest/config/__init__.py]: you can use `self.invocation_dir.join(x, abs=True)` and i believe the cast to str is not needed

Comment (blueyed) [reply src/_pytest/config/__init__.py]: The cast to string was needed for xdist.

Comment (RonnyPfannschmidt) [reply src/_pytest/config/__init__.py]: good point, then it is definitively needed for the line below where it sets `args = [self.invocation_dir]`

**Pull request merged**